File: src/main/java/ludwig/ide/SignatureListCell.java
package ludwig.ide;

import javafx.scene.control.ListCell;
import ludwig.model.Node;
import ludwig.utils.NodeUtils;

class SignatureListCell extends ListCell<Node<?>> {
    @Override
    protected void updateItem(Node<?> item, boolean empty) {
        super.updateItem(item, empty);

        setText((!empty && item != null) ? NodeUtils.signature(item) : "");
    }
}

File: src/main/java/ludwig/ide/SignatureListCell.kt
package ludwig.ide

import javafx.scene.control.ListCell
import ludwig.model.Node
import ludwig.utils.NodeUtils

internal class SignatureListCell : ListCell<Node<*>>() {
    override fun updateItem(item: Node<*>?, empty: Boolean) {
        super.updateItem(item, empty)

        text = if (!empty && item != null) NodeUtils.signature(item) else ""
    }
}
